packages/documentation/src/stories/components/switch/switch.stories.tsx

```diff
@@ -0,0 +1,242 @@
+      name: 'Disabled',
+      description: '<p>When set to `true`, disables the component\'s functionality and places it in a disabled state.</p><div className="alert alert-info alert-sm">There are accessibility issues with the disabled state.<br/>Please read our <a href="/?path=/docs/foundations-accessibility--page#disabled-state">disabled state accessibility guide</a>.</div>',
+      control: {
+        type: 'boolean'
+      },
+      table: {
+        category: 'States'
+      }
+    },
+    validation: {
+      name: 'Validation',
+      description: 'Controls the validation state appearance of the component.',
+      control: {
+        type: 'radio',
+        labels: {
+          'null': 'Default',
+          'is-valid': 'Valid',
+          'is-invalid': 'Invalid'
+        }
+      },
+      options: [
+        'null',
+        'is-valid',
+        'is-invalid'
+      ],
+      table: {
+        category: 'States'
+      }
+    },
+    validFeedback: {
+      name: 'Valid Feedback',
+      description: 'Text to show when the component is in an valid state.',
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'States'
+      }
+    },
+    invalidFeedback: {
+      name: 'Invalid Feedback',
+      description: 'Text to show when the component is in an invalid state.',
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'States'
+      }
+    }
+  }
+} as Meta;
+
+function toggle (args: Args, updateArgs: Function) {
+  updateArgs({ checked: !args.checked });
+}
+
+
+const Template = (args: Args, context: StoryContext<ReactFramework, Args>) => {
+  const [_, updateArgs] = useArgs();
+
+  const id = `ExampleSwitch_${context.name}`;
+  const classes = [
+    'form-check-input',
+    args.validation
+  ].filter(c => c && c !== 'null').join(' ');
+
+  const useAriaLabel = args.hiddenLabel;
+  const useLabelBefore = !useAriaLabel && ['before', 'both'].includes(args.labelPosition);
+  const useLabelAfter = !useAriaLabel && ['after', 'both'].includes(args.labelPosition);
+  const labelBefore: (JSX.Element | null) = useLabelBefore ? <label key="label-before" htmlFor={ id } className="form-check-label order-first">{ args.labelBefore }</label> : null;
+  const labelAfter: (JSX.Element | null) = useLabelAfter ? <label key="label-after" htmlFor={ id } className="form-check-label">{ args.labelAfter }</label> : null;
+
+  const contextuals: (JSX.Element | null)[] = [
+    args.validation === 'is-valid' ? <p key="valid" className="valid-feedback">{ args.validFeedback }</p> : null,
+    args.validation === 'is-invalid' ? <p key="invalid" className="invalid-feedback">{ args.invalidFeedback }</p> : null
+  ];
+
+  const control = <input
+    key="control"
+    id={ id }
+    className={ classes }
+    type="checkbox"
+    role="switch"
+    checked={ args.checked }
+    disabled={ args.disabled }
+    aria-label={ useAriaLabel ? `${args.labelBefore} ${args.labelAfter}` : undefined }
```

Comment:
This label reads as "check box checked Off On", which doesn't seem correct from an accessibility perspective.
`useAriaLabel ? args.labelAfter : undefined` might be more appropriate.

Reply:
I changed the code, so the aria-label always shows the current state (either `on` or `off`).